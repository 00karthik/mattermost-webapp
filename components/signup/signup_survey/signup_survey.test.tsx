// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import SignupSurvey from 'components/signup/signup_survey/signup_survey';
import {browserHistory} from 'utils/browser_history';
import {StoragePrefixes, SurveyTypes, SignupSurveyTelemetryEvents} from 'utils/constants';
import RadioSetting from '../../widgets/settings/radio_setting';

describe('components/SignupSurvey', () => {
    const defaultProps = {
        location: {
            state: {
                next: '/team-name',
            },
        },
        currentUserId: 'abc123',
        currentUserRoles: 'system_admin system_user',
        signupSurveyUserId: 'abc123',
        siteName: 'Mattermost',
        customDescriptionText: '',
        diagnosticsEnabled: true,
        actions: {
            removeGlobalItem: jest.fn().mockResolvedValue({data: true}),
        },
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <SignupSurvey
                {...defaultProps}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match redirect snapshot if not system_admin', () => {
        const wrapper = shallow(
            <SignupSurvey
                {...defaultProps}
                currentUserRoles='system_user'
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match redirect snapshot if not survey user', () => {
        const wrapper = shallow(
            <SignupSurvey
                {...defaultProps}
                currentUserId='xyz123'
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match redirect snapshot if diagnostics is disabled', () => {
        const wrapper = shallow(
            <SignupSurvey
                {...defaultProps}
                diagnosticsEnabled={false}
                currentUserId='xyz123'
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match redirect snapshot if next not provided', () => {
        const wrapper = shallow(
            <SignupSurvey
                {...defaultProps}
                location={{}}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should response to radio changes', () => {
        const wrapper = shallow<SignupSurvey>(
            <SignupSurvey
                {...defaultProps}
            />
        );

        wrapper.find(RadioSetting).prop('onChange')('', SignupSurveyTelemetryEvents.LONG_TERM);
        expect(wrapper.instance().state.serverPurpose).toBe(SignupSurveyTelemetryEvents.LONG_TERM);
    });

    it('should proceed to next after submission', async () => {
        browserHistory.push = jest.fn();

        const wrapper = shallow<SignupSurvey>(
            <SignupSurvey
                {...defaultProps}
            />
        );

        wrapper.find('#signupSurveyFinishButton').simulate('click', {preventDefault: jest.fn()});
        expect(defaultProps.actions.removeGlobalItem).toBeCalledWith(StoragePrefixes.SURVEY + SurveyTypes.SIGNUP);
        expect(browserHistory.push).toHaveBeenCalledWith(defaultProps.location.state.next);
    });
});
