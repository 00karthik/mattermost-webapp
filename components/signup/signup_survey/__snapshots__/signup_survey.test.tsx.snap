// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SignupSurvey should match redirect snapshot if diagnostics is disabled 1`] = `
<Redirect
  to="/team-name"
/>
`;

exports[`components/SignupSurvey should match redirect snapshot if next not provided 1`] = `
<Redirect
  to="/"
/>
`;

exports[`components/SignupSurvey should match redirect snapshot if not survey user 1`] = `
<Redirect
  to="/team-name"
/>
`;

exports[`components/SignupSurvey should match redirect snapshot if not system_admin 1`] = `
<Redirect
  to="/team-name"
/>
`;

exports[`components/SignupSurvey should match snapshot 1`] = `
<div>
  <div
    className="col-sm-12"
  >
    <form
      className="signup-team__container padding--less"
    >
      <img
        alt="signup team logo"
        className="signup-team-logo"
        src={null}
      />
      <SiteNameAndDescription
        customDescriptionText=""
        siteName="Mattermost"
      />
      <h3>
        <FormattedMessage
          defaultMessage="Almost Done"
          id="signup_survey.subtitle"
          values={Object {}}
        />
      </h3>
      <div
        className="form-group"
      >
        <div
          className="row"
        >
          <div
            className="col-sm-11"
          >
            <RadioSetting
              id="signup_survey_serverPurpose"
              inputClassName=""
              label={
                <FormattedMessage
                  defaultMessage="Before we wrap up, will this instance of Mattermost be a test server or something more permanent?"
                  id="signup_survey.server_purpose"
                  values={Object {}}
                />
              }
              labelClassName=""
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "text": <FormattedMessage
                      defaultMessage="Just taking a look around, this server probably won't live long"
                      id="signup_survey.server_purpose.short_term"
                      values={Object {}}
                    />,
                    "value": "survey_short_term",
                  },
                  Object {
                    "text": <FormattedMessage
                      defaultMessage="Setting this up for the long term"
                      id="signup_survey.server_purpose.long_term"
                      values={Object {}}
                    />,
                    "value": "survey_long_term",
                  },
                  Object {
                    "text": <FormattedMessage
                      defaultMessage="Not sure yet"
                      id="signup_survey.server_purpose.not_sure"
                      values={Object {}}
                    />,
                    "value": "survey_unsure",
                  },
                ]
              }
              value="survey_no_response"
            />
          </div>
        </div>
      </div>
      <div
        className="mt-8"
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="primary"
          disabled={false}
          id="signupSurveyFinishButton"
          onClick={[Function]}
          type="submit"
        >
          <FormattedMessage
            defaultMessage="Finish"
            id="signup_survey.finish"
            values={Object {}}
          />
        </Button>
      </div>
    </form>
  </div>
</div>
`;
